Zillow Features

hxliu33 opened the pull request:
takes input for type of home to search for (recently_sold, for_sale, or for_rent)

hxliu33 committed:
make it better

hxliu33 committed:
make it better

hxliu33 committed:
search for recently sold, for sale, or for rent

deoshao11 reviewed (changes requested):
Overall looks great! Awesome job Helen! Two small suggestions as below:

Please remove commented lines "#" in zillow.py's change, given those are not needed i assume?

Also the response.html should NOT be included in this PR, it is generated by the zillow.py. You can do git rm response.html to remove it. Even better, you can add *.html in the .gitignore file to ignore all *html in root dir when submit a PR.

hxliu33 committed:
Remove html files

unnecessary because these files are created by zillow.py

hxliu33 committed:
update zillow.py

take out commented out lines that weren't necessary anymore

hxliu33 commented:
> Overall looks great! Awesome job Helen! Two small suggestions as below:
> 
> Please remove commented lines "#" in zillow.py's change, given those are not needed i assume?
> 
> Also the response.html should NOT be included in this PR, it is generated by the zillow.py. You can do git rm response.html to remove it. Even better, you can add *.html in the .gitignore file to ignore all *html in root dir when submit a PR.

Hi Sunyi! I just fixed the files based on your comments (i.e. deleting response.html and deleting commented lines) in addition to creating another pull request to include *.html in the .gitignore file. I fixed the files directly in GitHub itself, but should I be doing it a different way? I wasn't sure if I was supposed to clone the whole repo again, make the edits there, and create a separate pull request in this branch in order to fix the files in response to your comments.

deoshao11 commented:
> > Overall looks great! Awesome job Helen! Two small suggestions as below:
> > Please remove commented lines "#" in zillow.py's change, given those are not needed i assume?
> > Also the response.html should NOT be included in this PR, it is generated by the zillow.py. You can do git rm response.html to remove it. Even better, you can add *.html in the .gitignore file to ignore all *html in root dir when submit a PR.
> 
> Hi Sunyi! I just fixed the files based on your comments (i.e. deleting response.html and deleting commented lines) in addition to creating another pull request to include *.html in the .gitignore file. I fixed the files directly in GitHub itself, but should I be doing it a different way? I wasn't sure if I was supposed to clone the whole repo again, make the edits there, and create a separate pull request in this branch in order to fix the files in response to your comments.

You are doing great and this is the way to go! I approved your PRs and you can click "merge pull request"'s drop down button and select "squash and merge" to merge the change into "master" branch.

The normal cycle of the software development (pull request) looks like this:
1.  `git checkout master` (make sure you are on the main branch)
2. `git fetch origin` (to sync up with remote git, here it is github)
3. `git pull` (apply all the latest changes since last sync to your local)
4. `git checkout -b <branch-name>` (branch out and create your own branch, so it won't impact anyone else working on the same project)
5. work on the changes in your branch
6. when you think u made some progress, do `git add .` && `git commit -m'<simple word>'` to commit your changes.
 7. repeat step 6 until you think the change is ready. Then do `git push origin <branch_name>` to surface your change into github remote
8. go to github website and create "pull request" from the branch
9. once you got feedback from reviewer, you can start from step 5 to 7, the PR will automatically update itself, so no need to create another PR.

This is a suggested workflow as a beginning, later on I may introduce some more fun procedures like CD/CI workflow(implies you can do git push more often and run your branch remotely with bunch of automated tests to ensure the change works or not breaking stuff). But for now, the steps should be good to go!
